Change how failure reporting is being printed to the screen

CrystalPea opened the pull request:
- change how the “FAILURES” header is presented, mainly change separator from an equal sign to a dot surrounded by whitespace;
- add indexing for each failure;
- format title of failing test and change separator from an equal sign to a dot surrounded by whitespace and move the title from the centre to the left;
- get rid of dashes separating snippets of code from test file and tested file within failure message for a test;
- add an empty line between report printouts for failing tests
- make methods called only from within the class private
- refactor code for SRP and inefficiencies
- change name from rspecify_pytest to EasyPytest

vixus0 commented on conftest.py:
```diff
@@ -80,33 +81,37 @@ def pytest_runtest_logstart(self, nodeid, location):
+
+    def _add_indentation_for_tests_list_item(self, report):
+        if len(report.nodeid.split("::")) >= 3:
+            return "    "
```
It's sort of hard to tell at a glance how many spaces you want here. You could define it programmatically: `return ' '*5`.

CrystalPea replied:
Great idea, I will do that now, thank you!

vixus0 commented on conftest.py:
```diff
@@ -80,33 +81,37 @@ def pytest_runtest_logstart(self, nodeid, location):
+        # get_formatted_test_title() introduced. It formats test title to make it more readable
         # checking for verbosity is turned off (verbose by default) as this plugin works best for verbose mode.
         # indentation and markup for test title introduced
         rep = report
```
Don't think this assignment is needed -- you could just use `report` directly.

CrystalPea replied:
Good call, I changed that now.

vixus0 commented on conftest.py:
```diff
@@ -119,16 +124,68 @@ def pytest_runtest_logreport(self, report):
         else:
             self.ensure_newline()
             if hasattr(rep, 'node'):
                 self._tw.write("[%s] " % rep.node.gateway.id)
```
You've got a mix of  `%` string formatting and `.format` string formatting. It would be good to make this consistent. I think `.format` is the favoured way but you could look at how it's done elsewhere in pytest, I suppose.

CrystalPea replied:
Good catch, `%` was used in original pytest code and `.format()` was my addition, so I changed it to `%` to be consistent :)

CrystalPea commented:
@vixus0 
Thank you so much for your review, it's been very helpful!

1. Good find with class name, I changed it to EasyTerminalReporter now, will commit soon

2. Going to read about docstrings now, this is a great idea actually as I saw them used and even participated in writing them but was not sure what they are, thank you!

As to why I comment on changes made in each function: it's because they are a mix of code that I copied from pytest and code that I changed or introduced, but in commits it looks like I made it all. I found if someone wants to review the code or to know what exactly has been changed and why when compared to original pytest code, they would need to compare the plugin with the pytest code and read the commits message, which is quite an involved and time-consuming process. It was even difficult for me after a month of break, so even for me, if I plan to change something in the future, it's a big help. I know that good code should be self-evident and not need comments, but I found in this particular project extensive comments have helped me a lot, mainly because it's a modification of a pre-existing code, which is done outside of the original code base.

3. As to file name, I think that changing it will actually be part of publishing the plugin. In the development stage I made all changes in a file named conftest.py, because this file changes local behaviour of pytest, and so I was able to use my plugin locally without publishing it this way. So after this pull request is merged, my next step will be changing that filename and doing other steps I need to do to publish my plugin :).
